### ISSUE TITLE
Confirm question crashes with some characters as sufix

### ISSUE THREAD

**Issue by rpbaltazar** [piotrmurach/tty-prompt#91]
### Are you in the right place?
* For issues or feature requests file a GitHub issue in this repository
* For general questions or discussion post in [Gitter](https://gitter.im/piotrmurach/tty)

### Describe the problem

When using `[`, `]` or `#` (at least) in the suffix string, it throws an error

### Steps to reproduce the problem
```
      commit_message = "\[#{finish_tag} #{current_story[:id]}\] #{current_story[:title]}"
      abort_commit = prompt.no?('Commit with standard message?') do |q|
        q.suffix commit_message
      end
      if abort_commit
        prompt.say 'Aborted'
      else
        GitUtils.commit commit_message
      end
```

### Actual behaviour
```
/Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt/confirm_question.rb:124:in `block in conversion': premature end of char-class: /^[Finishes 161574389] inventory-item-count-is-not-consistent-across-the-application|[$/i (RegexpError)
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt/converter_registry.rb:56:in `call'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt/converter_dsl.rb:19:in `convert'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt/question.rb:207:in `convert_result'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt/confirm_question.rb:79:in `render_question'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt/question.rb:120:in `render'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt/confirm_question.rb:66:in `call'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/tty-prompt-0.17.1/lib/tty/prompt.rb:341:in `no?'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bundler/gems/story_branch-4f8896d948f2/lib/story_branch/main.rb:67:in `story_finish'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bundler/gems/story_branch-4f8896d948f2/lib/story_branch/commands/finish.rb:16:in `execute'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bundler/gems/story_branch-4f8896d948f2/lib/story_branch/cli.rb:54:in `finish'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bundler/gems/story_branch-4f8896d948f2/exe/story_branch:14:in `<top (required)>'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bin/story_branch:23:in `load'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bin/story_branch:23:in `<main>'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bin/ruby_executable_hooks:15:in `eval'
	from /Users/rpbaltazar/.rvm/gems/ruby-2.4.3@airdrilling/bin/ruby_executable_hooks:15:in `<main>'
```

### Expected behaviour
What did you expect to happen?

### Describe your environment

* OS version: OSx 10.14 
* Ruby version: ruby 2.4.3p205
* TTY version: tty-prompt-0.17.1

**Comment by rpbaltazar** [comment]
After reading through my code for the millionth time, I realized that i shouldn't be using the suffix option in my particular use case.

Feel free to close it if you think the suffix should not have these characters.

**Comment by rpbaltazar** [comment]
I'll try to take a look at why is it happening and how to best fix it during today. I'm pretty sure it has to do with the regex comparison being used in the block of `confirm_question`, but I haven't dug into your code enough.

**Comment by piotrmurach** [comment]
Released `v0.17.2` with your patch :shipit:
